### PR TITLE
Feature: Back up data & 3dp profile migrate tip modal

### PR DESCRIPTION
**Pull request by wen-qing-255**

### CODE REVIEW

File: src/app/flux/printing/index.js [hunk 1]
```diff
@@ -2323,6 +2324,20 @@ export const actions = {
...
+    },
+
+    checkNewUser: () => (dispatch) => {
+        api.checkNewUser().then((res) => {
+            const isNewUser = res?.body?.isNewUser;
+            dispatch(actions.updateState({
+                isNewUser
+            }));
+        }).catch((err) => {
+            console.log({ err });
```
Comment (xiekun1992) [inline src/app/flux/printing/index.js:2336]: remove log

File: src/app/styles/global-font.styl [hunk 2]
```diff
@@ -123,6 +123,7 @@
     line-height: 16px;
     &:hover {
     color: $blue-3;
+    cursor: pointer;
```
Comment (xiekun1992) [inline src/app/styles/global-font.styl:126]: text shoud indent 2 spaces, make sure it is formatted

File: src/app/ui/pages/Printing.jsx [hunk 3]
```diff
@@ -309,7 +320,16 @@ function Printing({ location }) {
...
+    const handleCloseTipModal = () => {
+        setShowTipModal(false);
+        machineStore.set('readTip', true);
+    };
+    const openFolder = () => {
+        if (isElectron()) {
+            const ipc = window.require('electron').ipcRenderer;
+            ipc.send('open-recover-folder');
+        }
+    };
```
Comment (jane-rose) [inline src/app/ui/pages/Printing.jsx:332]: move this function outside component?

File: src/main.js [hunk 4]
```diff
@@ -293,6 +293,12 @@ const showMainWindow = async () => {
         addRecentFile(file);
     });
 
+    ipcMain.on('open-recover-folder', () => {
+        const userDataDir = app.getPath('userData');
+        console.log('userDataDir', userDataDir);
```
Comment (jane-rose) [inline src/main.js:298]: log

File: src/server/DataStorage.js [hunk 5]
```diff
@@ -235,6 +245,11 @@ class DataStorage {
          }
      }
 
+     async checkNewUser() {
+        const hasConfigDir = fs.existsSync(this.configDir);
+        config.set('isNewUser', !hasConfigDir);
```
Comment (jane-rose) [inline src/server/DataStorage.js:250]: as we discussed, make it a object

File: src/server/DataStorage.js [hunk 6]
```diff
@@ -127,22 +127,33 @@ class DataStorage {
...
                  const src = path.join(srcDir, file);
                  const dst = path.join(dstDir, file);
                  if (fs.statSync(src).isFile()) {
                      if (fs.existsSync(dst) && !overwriteTag) {
                          return;
                      }
-                     fs.copyFileSync(src, dst, (err) => {
-                         console.error('err', err);
-                     });
+                     if (needUpdateVersion) {
```
Comment (jane-rose) [inline src/server/DataStorage.js:142]: before `return`

**Pull request merged**